robot_model/include/moveit/robot_model/joint_model_group.h

```diff
@@ -485,6 +485,16 @@ class JointModelGroup
    */
   bool getEndEffectorTips(std::vector<std::string> &tips) const;
 
+  /**
+   * \brief Get one end effector tip, throwing an error if there ends up being more in the joint model group
+   *  This is a useful helper function because most planning groups (almost all) only have one tip
+   * \return pointer to LinkModel, or NULL on failure
+   */
+  const moveit::core::LinkModel* getOnlyOneEndEffectorTip() const;
+
+  /** \brief If this group is an end-effector, specify the foot type (if any, may be NONE) */
+  void setEndEffectorFootType(const JointModelGroupFootType &end_effector_foot_type);
```

Comment:
ooops, yes, that is part of my research code that leaked into this PR during a merge